### PR TITLE
Fix beam search

### PR DESCRIPTION
**Pull request by Adel-Moumen**
## What does this PR do?

This PR fixes several minor issues with the current Beam Search in unstable. 

I found one issue on `self.prefix_length` that was starting at `1` instead of `0` in the `CTCPrefixScorer`, and also, we were not blocking the `log_probs` of `blank_index` before computing the scores. 

Regarding the WER difference w.r.t `length_normalization`, the issue was due to: `final_scores = scores[index]`. Indeed, in unstable, we save the scores of an hyp that reach eos in a variable called `final_scores`, HOWEVER, doing so will not create a new variable but rather a reference to the variable `scores[index]`. What is the issue then ? The issue is that we are then setting the `scores[index]` to `-inf` because the hyp reached eos, and so `final_scores` will be equal to `-inf`... Why is it not happening in develop ? In develop, we are multiplying the `final_scores` with `length_rewarding` such as : `final_scores = scores[index] + self.length_rewarding * (timesteps + 1)`. Even if `self.length_rewarding` is equal to `0.0`, it will lead to a NEW variable even though we have the same final value, and thus, it will not be override by `-inf`....

I did the tests and it works. 

Example in python :
```python
import torch 

scores = torch.tensor([10.0, 20.0])
final_score = scores[0]
scores[0] = float('-inf')
final_score == float('-inf') # True...
```

An effective solution is to simply do `clone()`:

```python
import torch 

scores = torch.tensor([10.0, 20.0])
final_score = scores[0].clone()
scores[0] = float('-inf')
final_score == float('-inf') # False !
```

<details>
  <summary><b>Before submitting</b></summary>

- [x] Did you read the [contributor guideline](https://speechbrain.readthedocs.io/en/latest/contributing.html)?
- [x] Did you make sure your **PR does only one thing**, instead of bundling different changes together?
- [x] Did you make sure to **update the documentation** with your changes? (if necessary)
- [x] Did you write any **new necessary tests**? (not for typos and docs)
- [ x Did you verify new and **existing [tests](https://github.com/speechbrain/speechbrain/tree/develop/tests) pass** locally with your changes?
- [x] Did you list all the **breaking changes** introduced by this pull request?
- [x] Does your code adhere to project-specific code style and conventions?

</details>

## PR review

<details>
  <summary>Reviewer checklist</summary>

- [ ] Is this pull request ready for review? (if not, please submit in draft mode)
- [ ] Check that all items from **Before submitting** are resolved
- [ ] Make sure the title is self-explanatory and the description concisely explains the PR
- [ ] Add labels and milestones (and optionally projects) to the PR so it can be classified
- [ ] Confirm that the changes adhere to compatibility requirements (e.g., Python version, platform)
- [ ] Review the self-review checklist to ensure the code is ready for review

</details>

### CODE REVIEW

**Review by mravanelli — APPROVED** [review]

**Comment by mravanelli** [comment]
I did a full inference experiment with and without length normalization and everything works now. Thank you @Adel-Moumen for the fix.
